src/Microsoft.AspNetCore.Server.HttpSys/MessagePump.cs

```diff
@@ -40,10 +43,21 @@ public MessagePump(IOptions<HttpSysOptions> options, ILoggerFactory loggerFactor
             {
                 throw new ArgumentNullException(nameof(loggerFactory));
             }
-
             _options = options.Value;
             Listener = new HttpSysListener(_options, loggerFactory);
             _logger = LogHelper.CreateLogger(loggerFactory, typeof(MessagePump));
+
+            if (_options.Authentication.Schemes != AuthenticationSchemes.None)
+            {
+                var auth = authentication.FirstOrDefault();
+                if (auth == null)
+                {
+                    throw new InvalidOperationException("AddAuthentication() is required to use Authentication.");
```

Comment:
This is an interesting one. we know auth is on, and we have the user set on HttpContext.User. You don't really need the AuthN service unless you're mixing windows and anonymous access. How about throwing if AuthenticationSchemes != None && AllowAnonymous? That way you don't need the AuthN service for an app that handles AuthZ at the server level.